internal/v4/api.go

```diff
@@ -131,14 +131,15 @@ func ResolveURL(store *charmstore.Store, url *charm.Reference) error {
 	if url.Series != "" && url.Revision != -1 {
 		return nil
 	}
-	urls, err := store.ExpandURL(url)
-	if err != nil {
-		return errgo.Notef(err, "cannot expand URL")
+	entity, err := store.FindBestEntity(url, "_id", "promulgated-url")
+	if err != nil && errgo.Cause(err) != params.ErrNotFound {
```

Comment:
I think the NotFound error returned by FindBestEntity is probably fine for use here,
so we could just do:

     if err != nil { 
           return errgo.Mask(err, params.Is(params.ErrNotFound))
     }